src/rule/index.ts

```diff
@@ -193,20 +196,21 @@ const matchResponse = (
 };
 
 export const match = (api: a.Root, input: har.t): Result.Result => {
-  let result = matchRequest(api, input.request);
-
   const pathWithNoServer = removeServer(api, input.request.path);
-  if (result.apiSubtree[pathWithNoServer]) {
-    // no need to match response if we did not match path above
-    const path = result.apiSubtree[pathWithNoServer].x_name;
-    const operationToMatch = input.request.method;
-    result = matchResponse(
-      api.paths[path][operationToMatch],
-      input.response,
-      result,
-      path
-    );
+
+  let result = matchRequest(api, input.request);
+  if (!result.success) {
+    return result
   }
 
+  const path = result.apiSubtree[pathWithNoServer].x_name;
```

Comment:
This fails when for example `pathWithNoServer = /user/asd` and the key in result.apiSubtree is `/user/{username}`.

Reply:
We can call findPath again to find the key